cluster/images/provider-kafka-controller/Makefile

```diff
@@ -1,28 +1,36 @@
 # ====================================================================================
 # Setup Project
 
-PLATFORMS := linux_amd64 linux_arm64
 include ../../../build/makelib/common.mk
 
 # ====================================================================================
 #  Options
-IMAGE = $(BUILD_REGISTRY)/provider-kafka-controller-$(ARCH)
-OSBASEIMAGE = scratch
-include ../../../build/makelib/image.mk
+include ../../../build/makelib/imagelight.mk
 
 # ====================================================================================
 # Targets
 
 img.build:
 	@$(INFO) docker build $(IMAGE)
+	@$(MAKE) BUILD_ARGS="--load" img.build.shared
+	@$(OK) docker build $(IMAGE)
+
+
+img.publish:
+	@$(INFO) docker publish $(IMAGE)
+	@$(MAKE) BUILD_ARGS="--push" img.build.shared
+	@$(OK) docker publish $(IMAGE)
+
+img.build.shared:
 	@cp Dockerfile $(IMAGE_TEMP_DIR) || $(FAIL)
-	@cp -R ../../../package $(IMAGE_TEMP_DIR) || $(FAIL)
-	@cd $(IMAGE_TEMP_DIR) && $(SED_CMD) 's|BASEIMAGE|$(OSBASEIMAGE)|g' Dockerfile || $(FAIL)
-	@cd $(IMAGE_TEMP_DIR) && $(SED_CMD) 's|VERSION|$(VERSION)|g' package/crossplane.yaml || $(FAIL)
-	@cd $(IMAGE_TEMP_DIR) && find package -type f -name '*.yaml' -exec cat {} >> 'package.yaml' \; -exec printf '\n---\n' \; || $(FAIL)
-	@docker build $(BUILD_ARGS) \
-		--build-arg ARCH=$(ARCH) \
-		--build-arg TINI_VERSION=$(TINI_VERSION) \
+	@cp $(OUTPUT_DIR)/bin/$(OS)_$(ARCH)/provider $(IMAGE_TEMP_DIR) || $(FAIL)
```

Comment:
@stevendborrelli because we use `img.build.shared` for both `build` and `publish`, and we rebuild in `publish` (though most of it is cached), we have to actually copy the whole binary output directory, then add the proper ARCH in the Dockerfile itself (ref: https://github.com/crossplane/crossplane/blob/1f84012248a350b479a575214c17af5fe183138b/cluster/images/crossplane/Makefile#L26)